nextpnr.dockerfile

```diff
@@ -110,3 +138,6 @@ RUN cd /tmp/nextpnr/build \
 
 FROM base AS all
```

Comment:
This last "all" stage should include all `nextpnr-*` artifacts: `nextpnr-ice40`, `nextpnr-ecp5`, `nextpnr-gowin` and `nextpnr-generic`. Maybe it would be desirable to change this to build generic only, since others are built before:

```dockerfile
#---

FROM build AS build-all

RUN cd /tmp/nextpnr/build \
 && cmake .. \
   -DARCH=generic \
   -DBUILD_GUI=OFF \
   -DBUILD_PYTHON=ON \
   -DUSE_OPENMP=ON \
 && make -j $(nproc) \
 && make DESTDIR=/opt/nextpnr install

#---

FROM base AS all
COPY --from=build-ice40 /opt/nextpnr /
COPY --from=build-ecp5 /opt/nextpnr /
COPY --from=build-gowin /opt/nextpnr /
COPY --from=build-generic /opt/nextpnr /
```

Note that this last image does not include icestorm, prjtrellis or apicula/apycula. We can later add another image for that, on top of this stage. But I think it's better to handle that after this PR is done.

Reply:
Updating `build-all` to `build-generic` was contributed by @se-bi. Adding gowin after (#11) should be more straightforward now.